### ISSUE TITLE
At the tip revision (29 March 10:18) intellisense in Managers is locking up

### ISSUE THREAD

**Issue by sno036** [APSIMInitiative/ApsimX#7893]
It is working in Report
Reverting back to to the revision on 23 March 17:09 fixed it
UI completely locks up. If in Debug mode then can stop from VS otherwise need task manager

**Comment by hol353** [comment]
@zur003 This problem seems to have happened following the update of some nuget packages in your merged PR #7756. It hangs on line 132 of CodeCompletionService.cs (the second time through):

~~~c#
   CompletionList completionList = await service.GetCompletionsAsync(document, offset);
~~~

It is interesting that intellisense works in report but not manager scripts.

**Comment by zur003** [comment]
@hol353 - I don't really understand the completion stuff very well, but it appears that the ManagerPresenter, unlike other Presenters making use of Gtk's SourceView editor, does not make use of the editor's "ContextItemsNeeded" handler, but instead uses a Gtk "show-completion" event. So it's not too surprising that the problem only occurs with Manger scripts. 

This is a rather nasty bug. At this point, I don't have any insight into what is going wrong. PR #7756 didn't change anything at the Gtk level, but did update the stuff related to Roslyn, so that is likely to be where the problem lies.
